### PR TITLE
Implement Pop Last Layer on Sequential Model

### PR DESCRIPTION
**Pull request by albertomontesg**
Implement the feature to pop the last layer of the stack at the sequential model.
Now it would be only necessary to call one method to pop the last layer of an existing model:
```python
model.pop_layer()
```
Added to test and documented.
This comes from the proposal at #2371 made by @joelthchao

### CODE REVIEW

**Comment by fchollet** [comment]
This should achieve the same result:

```python
sequential_model.layers.pop()
sequential_model.add(new_layer)
```

I don't see the need to add a custom method for this.

**Comment by joelthchao** [comment]
@fchollet `model.layers.pop()` does not maintain `model.output` if no new layer is added.

**Comment by albertomontesg** [comment]
@fchollet A user case to use `model.pop_layer()` would be if you have a VGG network but then only want to extract the features after the convolutional network, so the last Dense layers should be remove and no other layer is added.

**Comment by albertomontesg** [comment]
It would be possible to merge it @fchollet ?

**Comment by fchollet** [comment]
I think this should be made available as an external function (not a method) via a Github Gist. Something like `pop_last_layer(model)` or `replace_last_layer(model, new_layer)`. So that users have a reference implementation if they want to do it.

But I don't think it is useful to have it as part of the codebase. Having less features/code in the codebase is one of our main goals.

**Comment by adityapatadia** [comment]
@fchollet Keras v1.0 made lot of breaking changes and documentation also does not reflect it. Apart from that, there is no changelog which accurately describes those changes. Some layers like LRN were also removed which makes it hard to implement older models like VGG-F. 

It would be great if we we can prevent such changes in future and document each change well. Frankly, the changes without much documentation resulted in too many trial-error loops and lot of frustration.

**Comment by dolaameng** [comment]
Vote +1 for the merge, because 
1. calling `model.layers.pop()` and `model.add()` doesn't work with `utils.visualization_util` and etc. 
2. using features from intermediate layers is so common in practice and "partial weight-loading" as described in [here](https://gist.github.com/fchollet/f35fbc80e066a49d65f1688a7e99f069) is not straightforward.
3. I am OK with the implementation either being an external function or a model method.

**Comment by nanoix9** [comment]
Agree with @dolaameng and Vote +1
I met the same problem recently which cost me almost a whole day util I found this issue by luck. It is necessary to make a dedicated function for popping layers.

**Comment by fchollet** [comment]
Ok, you guys convinced me that there is demand for this feature. I have added it. You can now `pop()` your `Sequential` models.

**Comment by alyato** [comment]
@albertomontesg  I wanna pop last layer and add new layer using the keras function api, not Seq model.

> This comes from the proposal at #2371 made by joelthchao

i rewrite my code,like this 

> x = Dense(4096, activation='relu', name='fc2')(x)
x = Dense(1000, activation='softmax', name='predictions')(x)
model = Model(img_input, x)
pop_layer(model)
x = Dense(8, activation='softmax', name='predictions')(x)
return model

Is it right?
